pyodide-build/pyodide_build/pypabuild.py

```diff
@@ -98,8 +98,17 @@ def _build_in_isolated_env(
         symlink_unisolated_packages(env)
         install_reqs(env, builder.build_system_requires)
         installed_requires_for_build = False
+        # "--build-option" contains custom options that only apply to current package,
+        # passing it to the dependencies may cause error.
+        # e.g., lightgbm accepts "--nomp" as one of its setup.py arguments,
+        # but "--nomp" isn't supported by its dependencies such as numpy, scipy, etc.
+        reqs_config_settings = {
+            k: v for k, v in config_settings.items() if k != "--build-option"
```

Comment:
Is our `backend-flags` propagating to dependencies? That seems very weird.

Reply:
Ah, okay we are passing backend_flags in `builder.get_requires_for_build`. I think we should only pass it to `builder.build` and not  `builder.get_requires_for_build` @hoodmane?

Reply:
I certainly didn't mean for it to propagate to dependencies!

Reply:
> I certainly didn't mean for it to propagate to dependencies!

Okay, thanks for the confirmation. @yzhliu Could you remove all backend-flags(config_settings) passed to `builder.get_requires_for_build`? Thanks for catching this out!

Reply:
sure, I'll do a sweep over the related code.

Reply:
@ryanking13 seems here are the only two usage of `builder.get_requires_for_build` in the code base.

Reply:
Yes, there are two appearances of `builder.get_requires_for_build`.

Reply:
I changed both of them, is the PR good to go? or did I miss anything else?

Reply:
Is it pushed? I can't notice the change and it looks like the last push you made is 3 days ago before the above comments.

Reply:
no the original commit already has the two changes,
https://github.com/pyodide/pyodide/pull/3138/files#diff-8a5df82b2fccc945f9e52b43277a75fa9e7a3f3ea1f5b03725b97d186f84225fR110
https://github.com/pyodide/pyodide/pull/3138/files#diff-8a5df82b2fccc945f9e52b43277a75fa9e7a3f3ea1f5b03725b97d186f84225fR122